src/dialogs/sidebar/dialog-edit-sidebar.ts

```diff
@@ -0,0 +1,159 @@
+import "@material/mwc-linear-progress/mwc-linear-progress";
+import { mdiClose } from "@mdi/js";
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog-header";
+import "../../components/ha-icon-button";
+import "../../components/ha-items-display-editor";
+import type { DisplayValue } from "../../components/ha-items-display-editor";
+import "../../components/ha-md-dialog";
+import type { HaMdDialog } from "../../components/ha-md-dialog";
+import { computePanels, PANEL_ICONS } from "../../components/ha-sidebar";
+import type { HomeAssistant } from "../../types";
+import type { EditSidebarDialogParams } from "./show-dialog-edit-sidebar";
+
+@customElement("dialog-edit-sidebar")
+class DialogEditSidebar extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _open = false;
+
+  @query("ha-md-dialog") private _dialog?: HaMdDialog;
+
+  @state() private _order: string[] = [];
+
+  @state() private _hidden: string[] = [];
+
+  private _saveCallback?: (order: string[], hidden: string[]) => void;
+
+  public async showDialog(params: EditSidebarDialogParams): Promise<void> {
+    this._open = true;
+
+    const storedOrder = localStorage.getItem("sidebarPanelOrder");
+    const storedHidden = localStorage.getItem("sidebarHiddenPanels");
```

Comment:
Why dont we use the `storage` decorator for this?

Reply:
Or pass the value when opening the dialog (I guess we then have to fetch them on the profile page...)

Reply:
Yes that was the reason that I don't have to get it in the profile page. I can add the `storage` decorator.

Reply:
It is ok like this, leave it. We don't need the decorator as we only fetch it anyway.